data-science-onramp/data-cleaning/clean.py

```diff
@@ -0,0 +1,179 @@
+import os
+import sys
+import re
+import datetime
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import UserDefinedFunction, lit
+from pyspark.sql.types import StringType, IntegerType, FloatType
+
+
+PROJECT_ID = sys.argv[1]
+BUCKET_NAME = sys.argv[2]
+TABLE = f'{PROJECT_ID}.new_york_citibike_trips.RAW_DATA'
+
+def trip_duration(duration):
+    '''Convert trip duration to seconds.'''
+    if duration:
+        number = re.match('\d*.\d', duration)
+        if number:
+            number = float(number[0])
+        else:
+            return None
+
+        if 's' in duration:
+            return int(number)
+        elif 'm' in duration:
+            return int(number * 60)
+        elif 'h' in duration:
+            return int(number * 60 * 60)
+        else:
+            if number < 0:
+                return None
+            else:
+                return int(number)
+
+def station_name(name):
+    '''Replaces '/' with '&'.'''
+    if name:
+        return name.replace('/', '&')
+
+def user_type(user):
+    '''Converts user type to 'Subscriber' or 'Customer'.'''
+    if user:
+        if user.lower().startswith('sub'):
+            return 'Subscriber'
+        elif user.lower().startswith('cust'):
+            return 'Customer'
+        else:
+            return None
+
+def gender(s):
+    '''Converts gender to 'Male' or 'Female' or '''
+    if s:
+        if s.lower().startswith('m'):
+            return 'Male'
+        elif s.lower().startswith('f'):
+            return 'Female'
+        else:
+            return None
+
+def angle(a):
+    '''Converts DMS notation to angles.'''
+    if a:
+        dms = re.match('(-?\d*).(-?\d*)\'(-?\d*)"', a)
```

Comment:
Does this check if there is a degrees symbol? You may want to check for it using the unicode value: "\u00B0"

Reply:
This checks for any single character. I was going to use the unicode value but I couldn't get it working well using re. I thought this was a fair compromise because there are no other instances in the dataset that this particular regex would match.